homeassistant/components/unifi/sensor.py

```diff
@@ -132,11 +133,27 @@ def async_device_outlet_supported_fn(controller: UniFiController, obj_id: str) -
     return controller.api.devices[obj_id].outlet_ac_power_budget is not None
 
 
+@callback
+def async_client_is_connected_fn(controller: UniFiController, obj_id: str) -> bool:
+    """Check if client was last seen recently."""
+    client = controller.api.clients[obj_id]
+
+    if (
+        dt_util.utcnow() - dt_util.utc_from_timestamp(client.last_seen or 0)
+        > controller.option_detection_time
+    ):
+        return False
+
+    return True
+
+
 @dataclass(frozen=True)
 class UnifiSensorEntityDescriptionMixin(Generic[HandlerT, ApiItemT]):
     """Validate and load entities from different UniFi handlers."""
 
     value_fn: Callable[[UniFiController, ApiItemT], datetime | float | str | None]
+    should_check_heartbeat: bool
+    is_connected_fn: Callable[[UniFiController, str], bool]
```

Comment:
```suggestion
    is_connected_fn: Callable[[UniFiController, str], bool] | None
```

Lets simplify, if function is None no heartbeat. Maybe possible to set None as default value and not assign it to all initiations?

Reply:
I had to move the is_connected_fn property to the UnifiSensorEntityDescription class to force the order of default vs non default arguments (similar to what is done in switch.py).
Possible other options:
- also give a default value of None to value_fn (and an additional check before calling it)
- remove the UnifiSensorEntityDescriptionMixin class completely